### DISCUSSION TITLE
Support for other MIN series models

### DISCUSSION THREAD

**Discussion by simar7** [johanmeijer/grott#501] [Q&A]
Hello! Thanks for the project.

I was wondering if other MIN series models are supported? I see the list [here](https://github.com/johanmeijer/grott#compatibility) but was curious if MIN 7600 TL-XH-US was supported or not?

**Comment by johanmeijer** [comment]
I do think it should be supported, but we will now for sure if we test it. Try to run Grott beta version with invtype=min.

**Reply by typxxi** [reply]
You will need one MIN whatsoever with a TL-XH version which is the high voltage hybrid inverter who confirms that the MIN is working.
Min 4600 TL for example is from the same family in general (MiN TL are 2 mppt inverters) but without hybrid option.

Afaik MIN is working but not sure about Hybrid and especiall 7600 which has not being sold in the EU I bet cause for example we have a limit for 1 phase inverters of 4600 Watt, but I have known the MiN 6000 as the biggest one cause in some EU states 6000 W is the limit for 1 phase. I was not aware that a MiN 7600 even exists cause it was not published in our listings which ended with a MiN 6000 and a * which explains that this was not sold here due to the 4600 W limitations.

In general the MiN4600 and MiN6000 have the same hardware and the only difference is the firmware. But I guess the MiN7600 will be a different least regarding dimensions and an even bigger passive alu cooling block. 
I am still playing around with grott but not using it in a productive way but it seem to work for all the MiC 600 to MiC 2000, then the MiC 3000 and MiN 4600 which we have and also a SPH4600 which is a single phase hybrid with a low voltage battery (48 V nominal, here 51,2 V with 16s LiFePo4 cells). But I think I have read somewhere that someone had a MIN hybrid running in Homeassistant.

People tend to use the software but not reporting back if it is working or what is working and what not. Therefore I am testing here carefully our family of 7 inverters and hope to report in May or so that all are working quite well.

**Reply by RickQureshi** [reply]
Hello, I have the MIN 7600 TL-XH-US and unfortunately:

	 - ('192.168.40.39', 56368) has connected
	 - Growatt packet received:
		  <socket.socket fd=4, family=2, type=1, proto=0, laddr=('172.30.33.1', 44292), raddr=('35.86.62.182', 22015)>
	 - Grott - grottproxy - Invalid data record received, processing stopped for this record
	 - ('192.168.40.39', 56368) has disconnected
	 - ('192.168.40.39', 56369) has connected
	 - Growatt packet received:
		  <socket.socket fd=4, family=2, type=1, proto=0, laddr=('172.30.33.1', 44296), raddr=('35.86.62.182', 22015)>
	 - Grott - grottproxy - Invalid data record received, processing stopped for this record
	 - ('192.168.40.39', 56369) has disconnected
	 - ('192.168.40.39', 56370) has connected
	 - Growatt packet received:
		  <socket.socket fd=4, family=2, type=1, proto=0, laddr=('172.30.33.1', 41482), raddr=('35.86.62.182', 22015)>

-----------------------------------------------------------
 Add-on: Grott Beta branch (2.8)
 The Growatt inverter monitor with current HA plugin integrated
-----------------------------------------------------------
 Add-on version: 0.1.7
 You are running the latest version of this add-on.
 System: Home Assistant OS 16.3  (amd64 / generic-x86-64)
 Home Assistant Core: 2025.11.3
 Home Assistant Supervisor: 2025.12.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------

Grott settings:
_Generic:
	version:             	 2.8.3
	verbose:             	 True
	trace:               	 False
	config file:         	 /config/grott/grott.ini
	minrecl:             	 100
	decrypt:             	 True
	compat:              	 False
	invtype:             	 min <- have tried default and capital MIN (on stable), same error
        invtypemap:          	 {}     <- have tried { "inverter serial", "min"} (on stable), same error 
	include_all:         	 False
	blockcmd:            	 False
	noipf:               	 False
	time:                	 auto
	sendbuf:             	 True
	timezone:            	 local
	valueoffset:         	 6
	offset:              	 6
	inverterid:          	 automatic
	mode:                	 proxy
	grottip              	 default
	grottport            	 5279    <*** note this is set to 22015 in Home Assistant config yaml file

TCPDump:

16:33:57.005280 IP 192.168.40.39.26340 > 192.168.40.213.53: 32639+ A? gw-solar-dc.vidagrid.com. (42)
        0x0000:  0021 9bd8 0908 08b6 1fec 79e0 0800 4500  .!........y...E.
        0x0010:  0046 02c5 0000 ff11 e694 c0a8 2827 c0a8  .F..........('..
        0x0020:  28d5 66e4 0035 0032 928c 7f7f 0100 0001  (.f..5.2........
        0x0030:  0000 0000 0000 0b67 772d 736f 6c61 722d  .......gw-solar-
        0x0040:  6463 0876 6964 6167 7269 6403 636f 6d00  dc.vidagrid.com.
        0x0050:  0001 0001                                ....
16:33:57.014743 IP 192.168.40.39.56393 > 192.168.40.213.22015: Flags [S], seq 3440548797, win 5744, options [mss 1440], length 0
        0x0000:  0021 9bd8 0908 08b6 1fec 79e0 0800 4500  .!........y...E.
        0x0010:  002c 02c6 0000 ff06 e6b8 c0a8 2827 c0a8  .,..........('..
        0x0020:  28d5 dc49 55ff cd12 9bbd 0000 0000 6002  (..IU.........`.
        0x0030:  1670 1464 0000 0204 05a0 0000 0000 0000  .p.d............
16:33:57.023712 IP 192.168.40.39.56393 > 192.168.40.213.22015: Flags [.], ack 833644153, win 5744, length 0
        0x0000:  0021 9bd8 0908 08b6 1fec 79e0 0800 4500  .!........y...E.
        0x0010:  0028 02c7 0000 ff06 e6bb c0a8 2827 c0a8  .(..........('..
        0x0020:  28d5 dc49 55ff cd12 9bbe 31b0 6679 5010  (..IU.....1.fyP.
        0x0030:  1670 93d3 0000 0000 0000 0000 0000 0000  .p..............
16:33:57.027104 IP 192.168.40.39.56393 > 192.168.40.213.22015: Flags [P.], seq 0:5, ack 1, win 5744, length 5
        0x0000:  0021 9bd8 0908 08b6 1fec 79e0 0800 4500  .!........y...E.
        0x0010:  002d 02c8 0000 ff06 e6b5 c0a8 2827 c0a8  .-..........('..
        0x0020:  28d5 dc49 55ff cd12 9bbe 31b0 6679 5018  (..IU.....1.fyP.
        0x0030:  1670 4ff4 0000 6865 6c6c 6f00 0000 0000  .pO...hello.....
16:34:02.034149 IP 192.168.40.39.56393 > 192.168.40.213.22015: Flags [F.], seq 5, ack 1, win 5744, length 0
        0x0000:  0021 9bd8 0908 08b6 1fec 79e0 0800 4500  .!........y...E.
        0x0010:  0028 02c9 0000 ff06 e6b9 c0a8 2827 c0a8  .(..........('..
        0x0020:  28d5 dc49 55ff cd12 9bc3 31b0 6679 5011  (..IU.....1.fyP.
        0x0030:  1670 93cd 0000 0000 0000 0000 0000 0000  .p..............
16:34:02.042245 IP 192.168.40.39.56393 > 192.168.40.213.22015: Flags [.], ack 2, win 5743, length 0
        0x0000:  0021 9bd8 0908 08b6 1fec 79e0 0800 4500  .!........y...E.
        0x0010:  0028 02ca 0000 ff06 e6b8 c0a8 2827 c0a8  .(..........('..
        0x0020:  28d5 dc49 55ff cd12 9bc4 31b0 667a 5010  (..IU.....1.fzP.
        0x0030:  166f 93cd 0000 0000 0000 0000 0000 0000  .o..............
16:34:12.564720 ARP, Request who-has 192.168.40.39 tell 192.168.40.39, length 50
        0x0000:  ffff ffff ffff 08b6 1fec 79e0 0806 0001  ..........y.....
        0x0010:  0800 0604 0001 08b6 1fec 79e0 c0a8 2827  ..........y...('
        0x0020:  0000 0000 0000 c0a8 2827 0000 0000 0000  ........('......
        0x0030:  0000 0000 0000 0000 0000 0000 0000 0000  ................
16:34:28.038830 IP 192.168.40.39.20005 > 192.168.40.213.53: 35625+ A? gw-solar-dc.vidagrid.com. (42)
        0x0000:  0021 9bd8 0908 08b6 1fec 79e0 0800 4500  .!........y...E.
        0x0010:  0046 02cb 0000 ff11 e68e c0a8 2827 c0a8  .F..........('..

Note:  This inverter has a build in datalogger and I am using a DNS redirect (dnsmasq) to my Home Assistant server hence port 53 in the first record.  I've attached full Grott log files from both beta and stable.

Thanks for your work on Grott and your help!

[1ef934e0_grott-beta_2025-12-07T21-46-28.241Z.log](https://github.com/user-attachments/files/24020007/1ef934e0_grott-beta_2025-12-07T21-46-28.241Z.log)
[1ef934e0_grott_2025-12-07T21-45-33.344Z.log](https://github.com/user-attachments/files/24020008/1ef934e0_grott_2025-12-07T21-45-33.344Z.log)

**Reply by typxxi** [reply]
from my memory: i had also been for what ever reason a grott beta version which then had been no longer updated. 

I suggest to leave beta behind and install the proper one, the most recent to start from there. 

Here in germany and in most EU countries single phase systems are limited to 4,6 kW (germany) and UK 6 kW (in 2018  when we had been there), hence 7,6 kW is way beyond our limits allowed here in the EU and I guess also in the Netherlands. 

I hope that it does not need  a special fix caus afaik the whole Min 600 to min 2000 series is based on 1 hardware and castrated by software. Same for the Min 3000 to Min 6000 family but i have no clue about any Min 7600. 

Simply try to run that with the most recent min series like Min 6000 cause that is the biggest among the MIN 4600 family.

**Reply by RickQureshi** [reply]
Hello, thanks for your reply.  I've tried the Grott beta branch addon by eggguy and as I understand from the github page that addon was built with grott3.0 alpha.  Same invalid record result with inverter type set to auto, min, default.  I've also tried the Grott stable branch with the same results.  

As I've seen the same type of error leading to the development of GroBro, I've gone on that Github page and installed the EMQX MQTT broker, which also logs an invalid record before GroBro or even the certs come into play (from my understanding of things.  It's entirely possible I'm doing something wrong as it's a first attempt)  Does that mean that this inverter doesn't use MQTT?

Couple of things:

1.  The URL the built in datalogger connects to is gw-solar-dc.vidagrid.com with an IP address of 35.86.62.182.
2. The built in datalogger's serial number VCXXXXXXXXXX1824 corresponds to [Growatt ShineWiFi-X].  So the issue may be the built in datalogger rather than the model of inverter itself.
3. The port is 22015.  I haven't seen either this URL or port number mentioned anywhere in any of the forums.  The only Google hit on the URL was that it was for the Shiner app.
4.  Using a DNS redirect and IPTABLES PREROUTING rule and POSTROUTING masquerade I'm able to do a TCPDUMP while the inverter is connected the Growatt (now gw-solar-vidagrid.com) cloud.   Everything on the shiner web portal and app functions perfectly.  I'm attaching the raw TCPDUMP in case it helps shed any light on this.


[tcpdumprawforWireshark.txt](https://github.com/user-attachments/files/24465701/tcpdumprawforWireshark.txt)

Are these known issues?  Are there any plans for supporting this inverter/built in datalogger configuration with Grott?  Is this in fact a scheme similar to NEO/SPF-series inverters and NOAH/NEXA and better suited for GroBro?

Thanks for your help!
